src/SAML2/ReceivedAuthnRequest.php

```diff
@@ -42,22 +45,25 @@ private function __construct(SAML2AuthnRequest $request)
 
     public static function from(string $decodedSamlRequest): ReceivedAuthnRequest
     {
-        if (!is_string($decodedSamlRequest) || empty($decodedSamlRequest)) {
```

Comment:
In this case, we recieve a saml request from the binding (post or redirect) that is low level encoded string data. I can get into the original developer not wrapping that into a VO. Although it might be a good idea!

Reply:
It was more of a general thingy